plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergExceptions.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import java.io.FileNotFoundException;
+
+public final class IcebergExceptions
+{
+    private IcebergExceptions() {}
+
+    public static boolean isNotFoundException(Throwable failure)
+    {
+        // qualified name, as this is NOT the io.trino.spi.connector.NotFoundException
+        return failure instanceof org.apache.iceberg.exceptions.NotFoundException ||
```

Comment:
getCasualChain().stream().anyMatch ?

Reply:
makes sense

Reply:
I cleaned it up a bit more. PTAL @raunaqmorarka and @wendigo